File: SentToDo.Web/Models/ToDoHistoryEntry.cs
namespace SentToDo.Web.Models;

public class DbToDoHistoryEntry : ToDoHistoryEntry
{
    public ApplicationUser User { get; set; }
}

public class ToDoHistoryEntry
{
    public int Id { get; set; }
    public long Timestamp { get; set; }
    public ToDoTask? OldValue { get; set; }
    public ToDoTask? NewValue { get; set; }
    public HistoryAction Action { get; set; }
}

public enum HistoryAction
{
    Added,
    Deleted,
    Modified
}

File: SentToDo.Web/ClientApp/src/api/models/ToDoHistoryEntry.ts
/* istanbul ignore file */
/* tslint:disable */
/* eslint-disable */

import type { HistoryAction } from './HistoryAction';
import type { ToDoTask } from './ToDoTask';

export type ToDoHistoryEntry = {
    id?: number;
    timestamp?: number;
    oldValue?: ToDoTask;
    newValue?: ToDoTask;
    action?: HistoryAction;
};